### ISSUE TITLE
Improve exception safety with smart pointers

### ISSUE THREAD

**Issue by elfring** [tamlok/vnote#132]
Would you like to wrap any pointer data members (or variables) with the template class “[std::unique_ptr](https://en.wikipedia.org/wiki/Smart_pointer#unique_ptr "Description for the usage of smart pointers")”?

Update candidates:
- [VInsertImageDialog::setupUI](https://github.com/tamlok/vnote/blob/c9b1801a5a47e360109f973b63fc9f0ff5e40e54/src/dialog/vinsertimagedialog.cpp#L63)
- [VWaitingWidget::setupUI](https://github.com/tamlok/vnote/blob/c9b1801a5a47e360109f973b63fc9f0ff5e40e54/src/vwaitingwidget.cpp#L14)

**Comment by tamlok** [comment]
Hi, Qt utilizes `QObject` to maintain parent-child relationship among all the objects. An object is responsible for releasing all its children. Hence smart pointer is weird to use with subclass of `QObject`. But it is still valuable to use smart pointer in class which does not subclass `QObject`. I will change it incrementally in the future.

Thanks!

**Comment by elfring** [comment]
Would you like to use the class “[`QPointer`](https://doc.qt.io/qt-5/qpointer.html#details "Description for the usage of QPointer")” at more places?

**Comment by tamlok** [comment]
Yes, I think we could use that at more places. I will pay attention to it in future development and gradually replace more raw pointers with it.
